app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Api::V1::ReservationsController < ApplicationController
+  # GET /reservations
+  def index
+    @reservations = Reservation.includes(:room).where('user_id = ?', current_user)
+
+    render json: @reservations
+  end
+
+  # POST /rooms/1/rservation
+  def create
+    @reservation = Reservation.new(reservation_params)
+    @reservation.update(user: current_user, room: Room.find(params[:room_id]))
```

Comment:
[OPTIONAL] See my comment on destry method and see how something similar may apply, also consider possibly using `includes` method and avoid N+1 problem